### PR TITLE
Added ability to define hashtables in values

### PR DESCRIPTION
**Pull request by kalarani**
With this change we should be able to let the 'values' hash contain hashtables in it.

Sample usage:

```
values:
    simpleKey: simpleValue
    hashTable:
        key: value

install:
    ...
    spliceMap:
        "App.config":
            "//appSettings/add[@key='simpleKey']/@value":   "{{simpleKey}}"
            "//appSettings/add[@key='complexKey']/@value":   "{{hashTable.key}}"
```

### CODE REVIEW

File: CommonFunctions/Apply-Values.ps1 [hunk 1]
```diff
@@ -22,8 +22,18 @@ function Apply-Values($config, $values) {
 function Apply-StringValue($thisValue, $values) {
     [regex]::matches($thisValue, "{{.*?}}") | select -expand value |%{
         $origKey = $_
-        $valueKey = $_.Trim("{{}}")
-        $thisValue = $values["$valueKey"] | %{ $thisValue -replace $origKey, $_ }
+        $thisValue = Extract-Value-For $_ $values | %{ $thisValue -replace $origKey, $_ }
```
Comment (manojlds) [inline CommonFunctions/Apply-Values.ps1]: Not Powershell convention to name functions like `Extract-Value-For`, better use `Extract-ValueFor`?

**Comment by manojlds** [comment]
@kalarani Looks like an useful feature. Will merge it once I verify the changes. Thanks

File: CommonFunctions/Apply-Values.ps1 (hunk 1 above)
Comment (kalarani) [inline CommonFunctions/Apply-Values.ps1]: Sure!

**Comment by kalarani** [comment]
@manojlds any updates on this?

**Comment by manojlds** [comment]
@kalarani sorry, didn't have windows box to test. Thanks for the PR, merged. Welcome more contributions. Happy to add as contributor if necessary.

**Pull request merged**